psh/date/date.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Phoenix-RTOS
+ *
+ * date - print or set the system date and time
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <sys/time.h>
+#include <unistd.h>
+
+#include "../psh.h"
+
+
+enum mode { SKIP,
+	HELP,
+	PRINT,
+	WRITE };
+
+static void psh_date_info(void)
+{
+	printf("print/set the system date and time");
+}
+
+
+static void psh_date_help(const char *prog)
+{
+	printf("Usage: %s [-h] [-s EPOCH] [+FORMAT]\n", prog);
+	printf("  -h:  shows this help message\n");
+	printf("  -s:  set system time described by EPOCH (POSIX time format)\n");
+	printf("  FORMAT: string with POSIX date formatting characters\n");
+	printf("NOTE: FORMAT string not supported with '-s' option\n");
+}
+
+
+static int psh_date_print(char *format)
+{
+	time_t t;
+	struct tm *tinfo;
+	char defformat[] = "+%a, %d %b %y %H:%M:%S";
```

Comment:
`const char`

Reply:
Nope, it's assigned to char *format as a pointer

Reply:
Added. Thanks.